exchanges/okgroup/okgroup_wrapper.go

```diff
@@ -78,22 +79,27 @@ func (o *OKGroup) FetchOrderbook(p currency.Pair, assetType asset.Item) (resp or
 }
 
 // UpdateOrderbook updates and returns the orderbook for a currency pair
-func (o *OKGroup) UpdateOrderbook(p currency.Pair, assetType asset.Item) (resp orderbook.Base, err error) {
-	orderbookNew, err := o.GetSpotOrderBook(GetSpotOrderBookRequest{
-		InstrumentID: o.FormatExchangeCurrency(p, assetType).String(),
-	})
+func (o *OKGroup) UpdateOrderbook(p currency.Pair, a asset.Item) (orderbook.Base, error) {
+	if a == asset.Index {
+		return orderbook.Base{}, errors.New("no orderbooks for index")
+	}
+
+	orderbookNew, err := o.GetOrderBook(GetSpotOrderBookRequest{
+		InstrumentID: o.FormatExchangeCurrency(p, a).String(),
+	}, a)
 	if err != nil {
-		return
+		return orderbook.Base{}, err
```

Comment:
If you  move the `resp` declaration above here, you can then send that instead of an initialised struct